extensions/data-plane/data-plane-google-bigquery/src/main/java/org/eclipse/edc/connector/dataplane/gcp/bigquery/params/BigQueryRequestParamsProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright (c) 2024 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.gcp.bigquery.params;
+
+import org.eclipse.edc.spi.types.domain.transfer.DataFlowStartMessage;
+
+/**
+ * Provides {@link DataFlowStartMessage} methods to prepare start message parameters from a
+ * BigQuery transfer request.
+ */
+public interface BigQueryRequestParamsProvider {
```

Comment:
Why does this have to be an interface? it's not an injectable service class, and its only job seems to be converting `DataflowStartMessage` objects into BQ request params. It seems more like a lib class

Reply:
Converted to class and Impl.java removed.